mesa/datacollection.py

```diff
@@ -21,19 +22,23 @@
     * model_vars maps each reporter to a list of its values
     * tables maps each table to a dictionary, with each column as a key with a
       list as its value.
-    * _agent_records maps each model step to a list of each agents id
+    * _agent_records maps each model step to a list of each agent's id
       and its values.
+    * _agenttype_records maps each model step to a dictionary of agent types,
+      each containing a list of each agent's id and its values.
 
 Finally, DataCollector can create a pandas DataFrame from each collection.
 
 The default DataCollector here makes several assumptions:
     * The model has an agent list called agents
+    * The model has a dictionary of AgentSets called agents_by_type
     * For collecting agent-level variables, agents must have a unique_id
```

Comment:
do we need this part about assumptions. All this is is in `Model`. Why not say: "the DataCollector is designed to work with `Model` or any subclass thereof?